### PR TITLE
Alpha40 module dependency resolvers

### PR DESCRIPTION
**Pull request by Azaezel**

### CODE REVIEW

File: Templates/BaseGame/game/core/utility/scripts/module.cs [hunk 1]
```diff
@@ -69,10 +83,12 @@ function loadModuleMaterials(%moduleGroup)
    return "";
 }
 
-function SimSet::registerDatablock(%scopeSet, %datablockFilePath)
+function SimSet::registerDatablock(%scopeSet, %datablockFilePath, %isExclusive)
```
Comment (lukaspj) [inline Templates/BaseGame/game/core/utility/scripts/module.cs:86]:
```suggestion
function SimSet::registerDatablockFile(%scopeSet, %datablockFilePath, %isExclusive)
```

File: Templates/BaseGame/game/core/utility/scripts/module.cs [hunk 2]
```diff
@@ -89,6 +105,176 @@ function loadModuleMaterials(%moduleGroup)
...
+   }
+   //if we're not locked, or we are exclusive, go ahead and add it to the pile
+   //(ensures exclusives get re-added after that erasure)
+   if (!%locked || %isExclusive)
+       DatablockFilesList.add(%fullPath,%isExclusive);
+   if ($traceModuleCalls)
+      DatablockFilesList.echo();
+}
+
+function SimSet::unRegisterDatablock(%scopeSet, %datablockFilePath)
```
Comment (lukaspj) [inline Templates/BaseGame/game/core/utility/scripts/module.cs:140]:
```suggestion
function SimSet::unregisterDatablockFile(%scopeSet, %datablockFilePath)
```

**Review by lukaspj — CHANGES_REQUESTED** [review]

**Review by lukaspj — COMMENTED** [review]
Could we create a DatablockFileList subclass of SimSet? Seems a bit odd that this functionality is defined on all SimSet's and not just the relevant ones

**Comment by Azaezel** [comment]
modules are themselves simests, though yeah, would tend to agree it may make things clearer there.

**Pull request merged**